updatecli/updatecli.d/jenkins-controllers/tools-jdk11.yaml

```diff
@@ -24,7 +24,8 @@ sources:
       versionfilter:
         kind: regex
         # jdk-11.0.12+7(https://github.com/adoptium/temurin11-binaries/releases/tag/jdk-11.0.12%2B7) is OK
-        pattern: "^jdk-11.(\\d*).(\\d*).(\\d*)+(\\d*)$"
+        # jdk-11.0.22+7.1 (https://github.com/adoptium/temurin11-binaries/releases/tag/jdk-11.0.22%2B7.1) is Ok
+        pattern: ^jdk-11.*.(\d*)+(\d*)$
```

Comment:
You should look at https://github.com/jenkinsci/docker-agent/pull/761/files/f95d636942e2b4308f3f1726064c4cde3ef0dc49#r1520406646 and reuse the same behavior to avoid repeating the same problems we faced on the official Jenkins Docker agent images:

- With this pattern change, the "latest" version found is `11.0.22+7.1` which only provides binaries for macOS aarch64 (see https://github.com/adoptium/temurin11-binaries/releases/tag/jdk-11.0.22%2B7.1): we want to use `jdk-11.0.22+7 ` instead (e.g. dropping any "hotfix")